src/main/java/org/kaazing/netx/ws/internal/WebSocketExtensionFactory.java

```diff
@@ -24,23 +24,27 @@
 import java.util.Map;
 import java.util.ServiceLoader;
 
+import org.kaazing.netx.ws.internal.ext.WebSocketExtensionFactorySpi;
+import org.kaazing.netx.ws.internal.ext.WebSocketExtensionSpi;
 
-public final class WebSocketExtensionFactory {
 
+public final class WebSocketExtensionFactory {
     private final Map<String, WebSocketExtensionFactorySpi> factoriesRO;
 
+    private WebSocketExtensionFactory(Map<String, WebSocketExtensionFactorySpi> factories) {
+        this.factoriesRO = factories;
```

Comment:
When all other changes are applied, maybe take a look to see if you want to keep this separate or make part of `WebSocketFactory` directly.